pkg/imageset/updateinstaller.go

```diff
@@ -174,6 +174,12 @@ func (o *UpdateInstallerImageOptions) Run() (returnErr error) {
 		if cd.Spec.Platform.BareMetal != nil {
 			installerTagName = "baremetal-installer"
 		}
+		// Override annotation is allowed to override baremetal-installer too
+		if cd.Annotations != nil {
+			if override := cd.Annotations[constants.OverrideInstallerImageNameAnnotation]; override != "" {
+				installerTagName = override
+			}
+		}
 		installerImage, err = findImageSpec(is, installerTagName)
 		if err != nil {
 			return errors.Wrap(err, "could not get installer image")
```

Comment:
Reminder if we can surface this in the CD conditions and have it be specific that the alternate image override is not found.

Reply:
Tried this out just now, cd.status.conditions gets:
```
    - lastProbeTime: "2023-10-27T21:37:18Z"
      lastTransitionTime: "2023-10-27T21:36:36Z"
      message: 'could not get installer image: no image tag "somethingnotfound" exists
        in the release image'
      reason: InstallerImageResolutionFailed
      status: "True"
      type: InstallerImageResolutionFailed
```

which seems like exactly what we're looking for, yes?